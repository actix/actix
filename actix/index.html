<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Actix is an actor framework."><title>actix - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/SourceSerif4-Regular-46f98efaafac5295.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/FiraSans-Regular-018c141bf0843ffd.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/FiraSans-Medium-8f9a781e4970d388.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/SourceSerif4-Bold-a2c9cd1067f8b328.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../static.files/SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2"><link rel="stylesheet" href="../static.files/normalize-76eba96aa4d2e634.css"><link rel="stylesheet" href="../static.files/rustdoc-085727c29c2fa99e.css"><meta name="rustdoc-vars" data-root-path="../" data-static-root-path="../static.files/" data-current-crate="actix" data-themes="" data-resource-suffix="" data-rustdoc-version="1.74.0-nightly (203c57dbe 2023-09-17)" data-channel="nightly" data-search-js="search-5d3eaacf19ebf04f.js" data-settings-js="settings-74424d7eec62a23e.js" ><script src="../static.files/storage-fec3eaa3851e447d.js"></script><script defer src="../crates.js"></script><script defer src="../static.files/main-f6ec288034ea164c.js"></script><noscript><link rel="stylesheet" href="../static.files/noscript-5d8b3c7633ad77ba.css"></noscript><link rel="icon" href="https://actix.rs/favicon.ico"></head><body class="rustdoc mod crate"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="logo-container" href="../actix/index.html"><img src="https://actix.rs/img/logo.png" alt="logo"></a><h2></h2></nav><nav class="sidebar"><a class="logo-container" href="../actix/index.html"><img src="https://actix.rs/img/logo.png" alt="logo"></a><h2 class="location"><a href="#">Crate actix</a></h2><div class="sidebar-elems"><ul class="block"><li class="version">Version 0.13.1</li><li><a id="all-types" href="all.html">All Items</a></li></ul><section><ul class="block"><li><a href="#reexports">Re-exports</a></li><li><a href="#modules">Modules</a></li><li><a href="#structs">Structs</a></li><li><a href="#enums">Enums</a></li><li><a href="#traits">Traits</a></li><li><a href="#functions">Functions</a></li><li><a href="#types">Type Aliases</a></li><li><a href="#attributes">Attribute Macros</a></li><li><a href="#derives">Derive Macros</a></li></ul></section></div></nav><main><div class="width-limiter"><nav class="sub"><form class="search-form"><span></span><input class="search-input" name="search" aria-label="Run search in the documentation" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" title="help" tabindex="-1"><a href="../help.html">?</a></div><div id="settings-menu" tabindex="-1"><a href="../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../static.files/wheel-7b819b6101059cd0.svg"></a></div></form></nav><section id="main-content" class="content"><div class="main-heading"><h1>Crate <a class="mod" href="#">actix</a><button id="copy-path" title="Copy item path to clipboard"><img src="../static.files/clipboard-7571035ce49a181d.svg" width="19" height="18" alt="Copy item path"></button></h1><span class="out-of-band"><a class="src" href="../src/actix/lib.rs.html#1-182">source</a> · <button id="toggle-all-docs" title="collapse all docs">[<span>&#x2212;</span>]</button></span></div><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Actix is an actor framework.</p>
<p><a href="./trait.Actor.html">Actors</a> are
objects which encapsulate state and behavior, they communicate
exclusively by exchanging messages. Actix actors are implemented
on top of <a href="https://tokio.rs">Tokio</a>.  Multiple actors can run in
same thread. Actors can run in multiple threads using the
<a href="struct.Arbiter.html"><code>Arbiter</code></a> API. Actors exchange typed
messages.</p>
<h3 id="other-documentation"><a href="#other-documentation">Other Documentation</a></h3>
<ul>
<li><a href="https://actix.rs/docs/actix">User Guide</a></li>
<li><a href="https://discord.gg/NWpN5mmg3x">Community Chat on Discord</a></li>
</ul>
<h3 id="features"><a href="#features">Features</a></h3>
<ul>
<li>Async/Sync actors</li>
<li>Actor communication in a local/thread context</li>
<li>Using Futures for asynchronous message handling</li>
<li>Actor supervision</li>
<li>Typed messages (No <a href="https://doc.rust-lang.org/nightly/core/any/trait.Any.html" title="trait core::any::Any"><code>Any</code></a> type). Generic messages are allowed</li>
<li>Runs on stable Rust 1.68+</li>
</ul>
</div></details><h2 id="reexports" class="small-section-header"><a href="#reexports">Re-exports</a></h2><ul class="item-table"><li><div class="item-name" id="reexport.ActorFuture"><code>pub use crate::fut::<a class="trait" href="fut/future/trait.ActorFuture.html" title="trait actix::fut::future::ActorFuture">ActorFuture</a>;</code></div></li><li><div class="item-name" id="reexport.ActorFutureExt"><code>pub use crate::fut::<a class="trait" href="fut/future/trait.ActorFutureExt.html" title="trait actix::fut::future::ActorFutureExt">ActorFutureExt</a>;</code></div></li><li><div class="item-name" id="reexport.ActorStream"><code>pub use crate::fut::<a class="trait" href="fut/stream/trait.ActorStream.html" title="trait actix::fut::stream::ActorStream">ActorStream</a>;</code></div></li><li><div class="item-name" id="reexport.ActorStreamExt"><code>pub use crate::fut::<a class="trait" href="fut/stream/trait.ActorStreamExt.html" title="trait actix::fut::stream::ActorStreamExt">ActorStreamExt</a>;</code></div></li><li><div class="item-name" id="reexport.ActorTryFuture"><code>pub use crate::fut::<a class="trait" href="fut/try_future/trait.ActorTryFuture.html" title="trait actix::fut::try_future::ActorTryFuture">ActorTryFuture</a>;</code></div></li><li><div class="item-name" id="reexport.ActorTryFutureExt"><code>pub use crate::fut::<a class="trait" href="fut/try_future/trait.ActorTryFutureExt.html" title="trait actix::fut::try_future::ActorTryFutureExt">ActorTryFutureExt</a>;</code></div></li><li><div class="item-name" id="reexport.WrapFuture"><code>pub use crate::fut::<a class="trait" href="fut/future/trait.WrapFuture.html" title="trait actix::fut::future::WrapFuture">WrapFuture</a>;</code></div></li><li><div class="item-name" id="reexport.WrapStream"><code>pub use crate::fut::<a class="trait" href="fut/stream/trait.WrapStream.html" title="trait actix::fut::stream::WrapStream">WrapStream</a>;</code></div></li><li><div class="item-name" id="reexport.ArbiterService"><code>pub use crate::registry::<a class="trait" href="registry/trait.ArbiterService.html" title="trait actix::registry::ArbiterService">ArbiterService</a>;</code></div></li><li><div class="item-name" id="reexport.Registry"><code>pub use crate::registry::<a class="struct" href="registry/struct.Registry.html" title="struct actix::registry::Registry">Registry</a>;</code></div></li><li><div class="item-name" id="reexport.SystemRegistry"><code>pub use crate::registry::<a class="struct" href="registry/struct.SystemRegistry.html" title="struct actix::registry::SystemRegistry">SystemRegistry</a>;</code></div></li><li><div class="item-name" id="reexport.SystemService"><code>pub use crate::registry::<a class="trait" href="registry/trait.SystemService.html" title="trait actix::registry::SystemService">SystemService</a>;</code></div></li><li><div class="item-name" id="reexport.SyncArbiter"><code>pub use crate::sync::<a class="struct" href="sync/struct.SyncArbiter.html" title="struct actix::sync::SyncArbiter">SyncArbiter</a>;</code></div></li><li><div class="item-name" id="reexport.SyncContext"><code>pub use crate::sync::<a class="struct" href="sync/struct.SyncContext.html" title="struct actix::sync::SyncContext">SyncContext</a>;</code></div></li></ul><h2 id="modules" class="small-section-header"><a href="#modules">Modules</a></h2><ul class="item-table"><li><div class="item-name"><a class="mod" href="actors/index.html" title="mod actix::actors">actors</a></div><div class="desc docblock-short">Helper actors</div></li><li><div class="item-name"><a class="mod" href="clock/index.html" title="mod actix::clock">clock</a></div><div class="desc docblock-short">Utilities for tracking time.</div></li><li><div class="item-name"><a class="mod" href="dev/index.html" title="mod actix::dev">dev</a></div><div class="desc docblock-short">The <code>actix</code> prelude for library developers.</div></li><li><div class="item-name"><a class="mod" href="fut/index.html" title="mod actix::fut">fut</a></div><div class="desc docblock-short">Custom <code>Future</code> and <code>Stream</code> implementation with <code>Actix</code> support</div></li><li><div class="item-name"><a class="mod" href="io/index.html" title="mod actix::io">io</a></div></li><li><div class="item-name"><a class="mod" href="prelude/index.html" title="mod actix::prelude">prelude</a></div><div class="desc docblock-short">The <code>actix</code> prelude.</div></li><li><div class="item-name"><a class="mod" href="registry/index.html" title="mod actix::registry">registry</a></div><div class="desc docblock-short">Actors registry</div></li><li><div class="item-name"><a class="mod" href="sync/index.html" title="mod actix::sync">sync</a></div><div class="desc docblock-short">Sync Actors support</div></li><li><div class="item-name"><a class="mod" href="utils/index.html" title="mod actix::utils">utils</a></div></li></ul><h2 id="structs" class="small-section-header"><a href="#structs">Structs</a></h2><ul class="item-table"><li><div class="item-name"><a class="struct" href="struct.ActorResponse.html" title="struct actix::ActorResponse">ActorResponse</a></div><div class="desc docblock-short">A helper type for representing different types of message responses.</div></li><li><div class="item-name"><a class="struct" href="struct.Addr.html" title="struct actix::Addr">Addr</a></div><div class="desc docblock-short">The address of an actor.</div></li><li><div class="item-name"><a class="struct" href="struct.Arbiter.html" title="struct actix::Arbiter">Arbiter</a></div><div class="desc docblock-short">An Arbiter represents a thread that provides an asynchronous execution environment for futures
and functions.</div></li><li><div class="item-name"><a class="struct" href="struct.ArbiterHandle.html" title="struct actix::ArbiterHandle">ArbiterHandle</a></div><div class="desc docblock-short">A handle for sending spawn and stop messages to an <a href="struct.Arbiter.html" title="struct actix::Arbiter">Arbiter</a>.</div></li><li><div class="item-name"><a class="struct" href="struct.AtomicResponse.html" title="struct actix::AtomicResponse">AtomicResponse</a></div><div class="desc docblock-short">A specialized actor future holder for atomic asynchronous message handling.</div></li><li><div class="item-name"><a class="struct" href="struct.Context.html" title="struct actix::Context">Context</a></div><div class="desc docblock-short">An actor execution context.</div></li><li><div class="item-name"><a class="struct" href="struct.MessageResult.html" title="struct actix::MessageResult">MessageResult</a></div><div class="desc docblock-short">A helper type that implements the <a href="dev/trait.MessageResponse.html" title="trait actix::dev::MessageResponse"><code>MessageResponse</code></a> trait.</div></li><li><div class="item-name"><a class="struct" href="struct.Recipient.html" title="struct actix::Recipient">Recipient</a></div><div class="desc docblock-short">The <a href="prelude/struct.Recipient.html" title="struct actix::prelude::Recipient"><code>Recipient</code></a> type allows to send one specific message to an actor.</div></li><li><div class="item-name"><a class="struct" href="struct.Response.html" title="struct actix::Response">Response</a></div><div class="desc docblock-short">Helper type for representing different type of message responses</div></li><li><div class="item-name"><a class="struct" href="struct.SpawnHandle.html" title="struct actix::SpawnHandle">SpawnHandle</a></div><div class="desc docblock-short">A handle to a spawned future.</div></li><li><div class="item-name"><a class="struct" href="struct.Supervisor.html" title="struct actix::Supervisor">Supervisor</a></div><div class="desc docblock-short">Actor supervisor</div></li><li><div class="item-name"><a class="struct" href="struct.System.html" title="struct actix::System">System</a></div><div class="desc docblock-short">A manager for a per-thread distributed async runtime.</div></li><li><div class="item-name"><a class="struct" href="struct.SystemRunner.html" title="struct actix::SystemRunner">SystemRunner</a></div><div class="desc docblock-short">Runner that keeps a <a href="struct.System.html" title="struct actix::System">System</a>’s event loop alive until stop message is received.</div></li><li><div class="item-name"><a class="struct" href="struct.WeakAddr.html" title="struct actix::WeakAddr">WeakAddr</a></div><div class="desc docblock-short">A weakly referenced counterpart to <code>Addr&lt;A&gt;</code>.</div></li><li><div class="item-name"><a class="struct" href="struct.WeakRecipient.html" title="struct actix::WeakRecipient">WeakRecipient</a></div><div class="desc docblock-short">A weakly referenced counterpart to <code>Recipient&lt;M&gt;</code></div></li></ul><h2 id="enums" class="small-section-header"><a href="#enums">Enums</a></h2><ul class="item-table"><li><div class="item-name"><a class="enum" href="enum.ActorState.html" title="enum actix::ActorState">ActorState</a></div><div class="desc docblock-short">Actor execution state</div></li><li><div class="item-name"><a class="enum" href="enum.MailboxError.html" title="enum actix::MailboxError">MailboxError</a></div><div class="desc docblock-short">The errors that can occur during the message delivery process.</div></li><li><div class="item-name"><a class="enum" href="enum.Running.html" title="enum actix::Running">Running</a></div></li></ul><h2 id="traits" class="small-section-header"><a href="#traits">Traits</a></h2><ul class="item-table"><li><div class="item-name"><a class="trait" href="trait.Actor.html" title="trait actix::Actor">Actor</a></div><div class="desc docblock-short">Actors are objects which encapsulate state and behavior.</div></li><li><div class="item-name"><a class="trait" href="trait.ActorContext.html" title="trait actix::ActorContext">ActorContext</a></div><div class="desc docblock-short">Actor execution context.</div></li><li><div class="item-name"><a class="trait" href="trait.AsyncContext.html" title="trait actix::AsyncContext">AsyncContext</a></div><div class="desc docblock-short">Asynchronous execution context.</div></li><li><div class="item-name"><a class="trait" href="trait.Handler.html" title="trait actix::Handler">Handler</a></div><div class="desc docblock-short">Describes how to handle messages of a specific type.</div></li><li><div class="item-name"><a class="trait" href="trait.Message.html" title="trait actix::Message">Message</a></div><div class="desc docblock-short">Represent message that can be handled by an actor.</div></li><li><div class="item-name"><a class="trait" href="trait.StreamHandler.html" title="trait actix::StreamHandler">StreamHandler</a></div><div class="desc docblock-short">Stream handling for Actors.</div></li><li><div class="item-name"><a class="trait" href="trait.Supervised.html" title="trait actix::Supervised">Supervised</a></div><div class="desc docblock-short">Actors with the ability to restart after failure.</div></li></ul><h2 id="functions" class="small-section-header"><a href="#functions">Functions</a></h2><ul class="item-table"><li><div class="item-name"><a class="fn" href="fn.run.html" title="fn actix::run">run</a></div><div class="desc docblock-short">Starts the system and executes the supplied future.</div></li><li><div class="item-name"><a class="fn" href="fn.spawn.html" title="fn actix::spawn">spawn</a></div><div class="desc docblock-short">Spawns a future on the current thread as a new task.</div></li></ul><h2 id="types" class="small-section-header"><a href="#types">Type Aliases</a></h2><ul class="item-table"><li><div class="item-name"><a class="type" href="type.ResponseActFuture.html" title="type actix::ResponseActFuture">ResponseActFuture</a></div><div class="desc docblock-short">A specialized actor future for asynchronous message handling.</div></li><li><div class="item-name"><a class="type" href="type.ResponseFuture.html" title="type actix::ResponseFuture">ResponseFuture</a></div><div class="desc docblock-short">A specialized future for asynchronous message handling.</div></li></ul><h2 id="attributes" class="small-section-header"><a href="#attributes">Attribute Macros</a></h2><ul class="item-table"><li><div class="item-name"><a class="attr" href="attr.main.html" title="attr actix::main">main</a></div><div class="desc docblock-short">Marks async main function as the <code>actix</code> system entry-point.</div></li><li><div class="item-name"><a class="attr" href="attr.test.html" title="attr actix::test">test</a></div><div class="desc docblock-short">Marks async test functions to use the <code>actix</code> system entry-point.</div></li></ul><h2 id="derives" class="small-section-header"><a href="#derives">Derive Macros</a></h2><ul class="item-table"><li><div class="item-name"><a class="derive" href="derive.Message.html" title="derive actix::Message">Message</a></div></li><li><div class="item-name"><a class="derive" href="derive.MessageResponse.html" title="derive actix::MessageResponse">MessageResponse</a></div></li></ul></section></div></main></body></html>