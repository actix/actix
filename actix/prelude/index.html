<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="The `actix` prelude."><title>actix::prelude - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/SourceSerif4-Regular-46f98efaafac5295.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/FiraSans-Regular-018c141bf0843ffd.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/FiraSans-Medium-8f9a781e4970d388.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/SourceSerif4-Bold-a2c9cd1067f8b328.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../static.files/SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2"><link rel="stylesheet" href="../../static.files/normalize-76eba96aa4d2e634.css"><link rel="stylesheet" href="../../static.files/rustdoc-085727c29c2fa99e.css"><meta name="rustdoc-vars" data-root-path="../../" data-static-root-path="../../static.files/" data-current-crate="actix" data-themes="" data-resource-suffix="" data-rustdoc-version="1.74.0-nightly (203c57dbe 2023-09-17)" data-channel="nightly" data-search-js="search-5d3eaacf19ebf04f.js" data-settings-js="settings-74424d7eec62a23e.js" ><script src="../../static.files/storage-fec3eaa3851e447d.js"></script><script defer src="../../static.files/main-f6ec288034ea164c.js"></script><noscript><link rel="stylesheet" href="../../static.files/noscript-5d8b3c7633ad77ba.css"></noscript><link rel="icon" href="https://actix.rs/favicon.ico"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="logo-container" href="../../actix/index.html"><img src="https://actix.rs/img/logo.png" alt="logo"></a><h2></h2></nav><nav class="sidebar"><a class="logo-container" href="../../actix/index.html"><img src="https://actix.rs/img/logo.png" alt="logo"></a><h2 class="location"><a href="#">Module prelude</a></h2><div class="sidebar-elems"><section><ul class="block"><li><a href="#reexports">Re-exports</a></li><li><a href="#structs">Structs</a></li><li><a href="#enums">Enums</a></li><li><a href="#traits">Traits</a></li><li><a href="#types">Type Aliases</a></li></ul></section></div></nav><main><div class="width-limiter"><nav class="sub"><form class="search-form"><span></span><input class="search-input" name="search" aria-label="Run search in the documentation" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" title="help" tabindex="-1"><a href="../../help.html">?</a></div><div id="settings-menu" tabindex="-1"><a href="../../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../../static.files/wheel-7b819b6101059cd0.svg"></a></div></form></nav><section id="main-content" class="content"><div class="main-heading"><h1>Module <a href="../index.html">actix</a>::<wbr><a class="mod" href="#">prelude</a><button id="copy-path" title="Copy item path to clipboard"><img src="../../static.files/clipboard-7571035ce49a181d.svg" width="19" height="18" alt="Copy item path"></button></h1><span class="out-of-band"><a class="src" href="../../src/actix/lib.rs.html#80">source</a> · <button id="toggle-all-docs" title="collapse all docs">[<span>&#x2212;</span>]</button></span></div><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>The <code>actix</code> prelude.</p>
<p>The purpose of this module is to alleviate imports of many common actix
traits by adding a glob import to the top of actix heavy modules:</p>

<div class="example-wrap"><pre class="rust rust-example-rendered"><code><span class="kw">use </span>actix::prelude::<span class="kw-2">*</span>;</code></pre></div>
</div></details><h2 id="reexports" class="small-section-header"><a href="#reexports">Re-exports</a></h2><ul class="item-table"><li><div class="item-name" id="reexport.Condition"><code>pub use crate::utils::<a class="struct" href="../utils/struct.Condition.html" title="struct actix::utils::Condition">Condition</a>;</code></div><div class="desc docblock-short"><span class="stab deprecated" title="">Deprecated</span></div></li><li><div class="item-name" id="reexport.actors"><code>pub use crate::<a class="mod" href="../actors/index.html" title="mod actix::actors">actors</a>;</code></div></li><li><div class="item-name" id="reexport.dev"><code>pub use crate::<a class="mod" href="../dev/index.html" title="mod actix::dev">dev</a>;</code></div></li><li><div class="item-name" id="reexport.fut"><code>pub use crate::<a class="mod" href="../fut/index.html" title="mod actix::fut">fut</a>;</code></div></li><li><div class="item-name" id="reexport.ActorFuture"><code>pub use crate::fut::<a class="trait" href="../fut/future/trait.ActorFuture.html" title="trait actix::fut::future::ActorFuture">ActorFuture</a>;</code></div></li><li><div class="item-name" id="reexport.ActorFutureExt"><code>pub use crate::fut::<a class="trait" href="../fut/future/trait.ActorFutureExt.html" title="trait actix::fut::future::ActorFutureExt">ActorFutureExt</a>;</code></div></li><li><div class="item-name" id="reexport.ActorStream"><code>pub use crate::fut::<a class="trait" href="../fut/stream/trait.ActorStream.html" title="trait actix::fut::stream::ActorStream">ActorStream</a>;</code></div></li><li><div class="item-name" id="reexport.ActorStreamExt"><code>pub use crate::fut::<a class="trait" href="../fut/stream/trait.ActorStreamExt.html" title="trait actix::fut::stream::ActorStreamExt">ActorStreamExt</a>;</code></div></li><li><div class="item-name" id="reexport.ActorTryFuture"><code>pub use crate::fut::<a class="trait" href="../fut/try_future/trait.ActorTryFuture.html" title="trait actix::fut::try_future::ActorTryFuture">ActorTryFuture</a>;</code></div></li><li><div class="item-name" id="reexport.ActorTryFutureExt"><code>pub use crate::fut::<a class="trait" href="../fut/try_future/trait.ActorTryFutureExt.html" title="trait actix::fut::try_future::ActorTryFutureExt">ActorTryFutureExt</a>;</code></div></li><li><div class="item-name" id="reexport.WrapFuture"><code>pub use crate::fut::<a class="trait" href="../fut/future/trait.WrapFuture.html" title="trait actix::fut::future::WrapFuture">WrapFuture</a>;</code></div></li><li><div class="item-name" id="reexport.WrapStream"><code>pub use crate::fut::<a class="trait" href="../fut/stream/trait.WrapStream.html" title="trait actix::fut::stream::WrapStream">WrapStream</a>;</code></div></li><li><div class="item-name" id="reexport.io"><code>pub use crate::<a class="mod" href="../io/index.html" title="mod actix::io">io</a>;</code></div></li><li><div class="item-name" id="reexport.ArbiterService"><code>pub use crate::registry::<a class="trait" href="../registry/trait.ArbiterService.html" title="trait actix::registry::ArbiterService">ArbiterService</a>;</code></div></li><li><div class="item-name" id="reexport.SystemService"><code>pub use crate::registry::<a class="trait" href="../registry/trait.SystemService.html" title="trait actix::registry::SystemService">SystemService</a>;</code></div></li><li><div class="item-name" id="reexport.SyncArbiter"><code>pub use crate::sync::<a class="struct" href="../sync/struct.SyncArbiter.html" title="struct actix::sync::SyncArbiter">SyncArbiter</a>;</code></div></li><li><div class="item-name" id="reexport.SyncContext"><code>pub use crate::sync::<a class="struct" href="../sync/struct.SyncContext.html" title="struct actix::sync::SyncContext">SyncContext</a>;</code></div></li><li><div class="item-name" id="reexport.IntervalFunc"><code>pub use crate::utils::<a class="struct" href="../utils/struct.IntervalFunc.html" title="struct actix::utils::IntervalFunc">IntervalFunc</a>;</code></div></li><li><div class="item-name" id="reexport.TimerFunc"><code>pub use crate::utils::<a class="struct" href="../utils/struct.TimerFunc.html" title="struct actix::utils::TimerFunc">TimerFunc</a>;</code></div></li></ul><h2 id="structs" class="small-section-header"><a href="#structs">Structs</a></h2><ul class="item-table"><li><div class="item-name"><a class="struct" href="struct.ActorResponse.html" title="struct actix::prelude::ActorResponse">ActorResponse</a></div><div class="desc docblock-short">A helper type for representing different types of message responses.</div></li><li><div class="item-name"><a class="struct" href="struct.Addr.html" title="struct actix::prelude::Addr">Addr</a></div><div class="desc docblock-short">The address of an actor.</div></li><li><div class="item-name"><a class="struct" href="struct.Arbiter.html" title="struct actix::prelude::Arbiter">Arbiter</a></div><div class="desc docblock-short">An Arbiter represents a thread that provides an asynchronous execution environment for futures
and functions.</div></li><li><div class="item-name"><a class="struct" href="struct.ArbiterHandle.html" title="struct actix::prelude::ArbiterHandle">ArbiterHandle</a></div><div class="desc docblock-short">A handle for sending spawn and stop messages to an <a href="../struct.Arbiter.html" title="struct actix::Arbiter">Arbiter</a>.</div></li><li><div class="item-name"><a class="struct" href="struct.AtomicResponse.html" title="struct actix::prelude::AtomicResponse">AtomicResponse</a></div><div class="desc docblock-short">A specialized actor future holder for atomic asynchronous message handling.</div></li><li><div class="item-name"><a class="struct" href="struct.Context.html" title="struct actix::prelude::Context">Context</a></div><div class="desc docblock-short">An actor execution context.</div></li><li><div class="item-name"><a class="struct" href="struct.MessageResult.html" title="struct actix::prelude::MessageResult">MessageResult</a></div><div class="desc docblock-short">A helper type that implements the <a href="../dev/trait.MessageResponse.html" title="trait actix::dev::MessageResponse"><code>MessageResponse</code></a> trait.</div></li><li><div class="item-name"><a class="struct" href="struct.Recipient.html" title="struct actix::prelude::Recipient">Recipient</a></div><div class="desc docblock-short">The <a href="struct.Recipient.html" title="struct actix::prelude::Recipient"><code>Recipient</code></a> type allows to send one specific message to an actor.</div></li><li><div class="item-name"><a class="struct" href="struct.Response.html" title="struct actix::prelude::Response">Response</a></div><div class="desc docblock-short">Helper type for representing different type of message responses</div></li><li><div class="item-name"><a class="struct" href="struct.SpawnHandle.html" title="struct actix::prelude::SpawnHandle">SpawnHandle</a></div><div class="desc docblock-short">A handle to a spawned future.</div></li><li><div class="item-name"><a class="struct" href="struct.Supervisor.html" title="struct actix::prelude::Supervisor">Supervisor</a></div><div class="desc docblock-short">Actor supervisor</div></li><li><div class="item-name"><a class="struct" href="struct.System.html" title="struct actix::prelude::System">System</a></div><div class="desc docblock-short">A manager for a per-thread distributed async runtime.</div></li><li><div class="item-name"><a class="struct" href="struct.SystemRunner.html" title="struct actix::prelude::SystemRunner">SystemRunner</a></div><div class="desc docblock-short">Runner that keeps a <a href="../struct.System.html" title="struct actix::System">System</a>’s event loop alive until stop message is received.</div></li></ul><h2 id="enums" class="small-section-header"><a href="#enums">Enums</a></h2><ul class="item-table"><li><div class="item-name"><a class="enum" href="enum.ActorState.html" title="enum actix::prelude::ActorState">ActorState</a></div><div class="desc docblock-short">Actor execution state</div></li><li><div class="item-name"><a class="enum" href="enum.MailboxError.html" title="enum actix::prelude::MailboxError">MailboxError</a></div><div class="desc docblock-short">The errors that can occur during the message delivery process.</div></li><li><div class="item-name"><a class="enum" href="enum.Running.html" title="enum actix::prelude::Running">Running</a></div></li><li><div class="item-name"><a class="enum" href="enum.SendError.html" title="enum actix::prelude::SendError">SendError</a></div></li></ul><h2 id="traits" class="small-section-header"><a href="#traits">Traits</a></h2><ul class="item-table"><li><div class="item-name"><a class="trait" href="trait.Actor.html" title="trait actix::prelude::Actor">Actor</a></div><div class="desc docblock-short">Actors are objects which encapsulate state and behavior.</div></li><li><div class="item-name"><a class="trait" href="trait.ActorContext.html" title="trait actix::prelude::ActorContext">ActorContext</a></div><div class="desc docblock-short">Actor execution context.</div></li><li><div class="item-name"><a class="trait" href="trait.AsyncContext.html" title="trait actix::prelude::AsyncContext">AsyncContext</a></div><div class="desc docblock-short">Asynchronous execution context.</div></li><li><div class="item-name"><a class="trait" href="trait.ContextFutureSpawner.html" title="trait actix::prelude::ContextFutureSpawner">ContextFutureSpawner</a></div><div class="desc docblock-short">Helper trait which can spawn a future into the actor’s context.</div></li><li><div class="item-name"><a class="trait" href="trait.Handler.html" title="trait actix::prelude::Handler">Handler</a></div><div class="desc docblock-short">Describes how to handle messages of a specific type.</div></li><li><div class="item-name"><a class="trait" href="trait.Message.html" title="trait actix::prelude::Message">Message</a></div><div class="desc docblock-short">Represent message that can be handled by an actor.</div></li><li><div class="item-name"><a class="trait" href="trait.Stream.html" title="trait actix::prelude::Stream">Stream</a></div><div class="desc docblock-short">A stream of values produced asynchronously.</div></li><li><div class="item-name"><a class="trait" href="trait.StreamHandler.html" title="trait actix::prelude::StreamHandler">StreamHandler</a></div><div class="desc docblock-short">Stream handling for Actors.</div></li><li><div class="item-name"><a class="trait" href="trait.Supervised.html" title="trait actix::prelude::Supervised">Supervised</a></div><div class="desc docblock-short">Actors with the ability to restart after failure.</div></li></ul><h2 id="types" class="small-section-header"><a href="#types">Type Aliases</a></h2><ul class="item-table"><li><div class="item-name"><a class="type" href="type.RecipientRequest.html" title="type actix::prelude::RecipientRequest">RecipientRequest</a></div></li><li><div class="item-name"><a class="type" href="type.Request.html" title="type actix::prelude::Request">Request</a></div></li><li><div class="item-name"><a class="type" href="type.ResponseActFuture.html" title="type actix::prelude::ResponseActFuture">ResponseActFuture</a></div><div class="desc docblock-short">A specialized actor future for asynchronous message handling.</div></li><li><div class="item-name"><a class="type" href="type.ResponseFuture.html" title="type actix::prelude::ResponseFuture">ResponseFuture</a></div><div class="desc docblock-short">A specialized future for asynchronous message handling.</div></li></ul></section></div></main></body></html>