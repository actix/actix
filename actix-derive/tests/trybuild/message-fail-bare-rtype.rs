use actix::prelude::*;

#[derive(MessageResponse)]
struct Added(usize);

#[derive(Message)]
#[rtype(result = Added)]
struct Sum(usize, usize);

fn main() {}
