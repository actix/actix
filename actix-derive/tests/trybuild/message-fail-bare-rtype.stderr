error: Expect type
 --> tests/trybuild/message-fail-bare-rtype.rs:7:18
  |
7 | #[rtype(result = Added)]
  |                  ^^^^^
